File: generators/app/templates/java/app/src/main/java/org/onepointzero/androidstarterjava/infraestructure/DummyStarterDataRepository.java
package <%= packageName %>.infraestructure;

import android.content.Context;

import <%= packageName %>.R;
import <%= packageName %>.domain.StarterData;
import <%= packageName %>.domain.StarterDataRepository;

import io.reactivex.Single;

public class DummyStarterDataRepository implements StarterDataRepository {

    Context context;

    public DummyStarterDataRepository(Context context) {
        this.context = context;
    }

    @Override
    public Single<StarterData> getStarterData() {

        String appName = context.getResources().getString(R.string.app_name);

        StarterData data = new StarterData("Hello " + appName + "!");
        return Single.just(data);
    }
}

File: generators/app/templates/kotlin/app/src/main/java/org/onepointzero/androidstarterkotlin/infraestructure/DummyStarterDataRepository.kt
package <%= packageName %>.infraestructure

import android.content.Context

import io.reactivex.Single
import <%= packageName %>.R
import <%= packageName %>.domain.StarterData
import <%= packageName %>.domain.StarterDataRepository

class DummyStarterDataRepository(private var context: Context) : StarterDataRepository {

    override fun get(): Single<StarterData> {
        val appName = context.resources.getString(R.string.app_name)

        val data = StarterData("Hello $appName!")
        return Single.just<StarterData>(data)
    }
}
